azurerm/internal/services/synapse/client/client.go

```diff
@@ -52,9 +52,12 @@ func NewClient(o *common.ClientOptions) *Client {
 	}
 }
 
-func (client Client) AccessControlClient(workspaceName, synapseEndpointSuffix string) *accesscontrol.BaseClient {
+func (client Client) AccessControlClient(workspaceName, synapseEndpointSuffix string) (*accesscontrol.BaseClient, error) {
+	if client.synapseAuthorizer == nil {
+		return nil, fmt.Errorf("synapse service is not supported in current environment")
```

Comment:
can we make this a little more specific via:

```suggestion
		return nil, fmt.Errorf("Synapse is not supported in this Azure Environment")
```

(environment is ambiguous, and could refer to the Azure Env or another type of environment, such as the CLI environment)